test/run_e2e_test.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Terminal output colors
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+CYAN='\033[0;36m'
+NORM='\033[0m'
+# Defaults
+ARCH="aarch64"  # or "armhf"
+OS="ubuntu"     # or "debian"
+DISTRO="dashing"
+RMW="fastrtps"
+RESULT=1        # Default to failure
+
+cleanup(){
+  # Usage: cleanup CONTAINER_NAME RESULT
+  if [[ "$2" -eq 0 ]]; then
+    printf "%b%s%b\n" "$GREEN" "PASS" "$NORM"
+  else
+    printf "%b%s%b\n" "$RED" "FAIL" "$NORM"
+  fi
+  rm -rf /tmp/sysroot;
+  docker rm $1 >/dev/null 2>/dev/null;
+}
+
+setup(){
+  mkdir /tmp/sysroot
+  mkdir -p /tmp/sysroot/ros2_ws/src/
+  wget https://raw.githubusercontent.com/ros2/ros2/release-latest/ros2.repos --directory-prefix=/tmp/sysroot
+  vcs import /tmp/sysroot/ros2_ws/src < /tmp/sysroot/ros2.repos
+  mkdir -p /tmp/sysroot/qemu-user-static
+  cp /usr/bin/qemu-* /tmp/sysroot/qemu-user-static
+  # Get full directory name of the script no matter where it is being called from
+  DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+  cp $DIR/Dockerfile_workspace /tmp/sysroot/
+}
+
+# Parse options
+while [[ $# -gt 0 ]]
+do
+  key="$1"
+
+  case $key in
+      -a|--arch)
+      ARCH="$2"
+      shift
+      shift
+      ;;
+      -o|--os)
+      OS="$2"
+      shift
+      shift
+      ;;
+      -d|--distro)
+      DISTRO="$2"
+      shift
+      shift
+      ;;
+      -r|--rmw)
+      RMW="$2"
+      shift
+      shift
+      ;;
+      *)    # Unknown option
+      echo "Unknown Flag $1"
+      exit 1
+      shift
+      ;;
+  esac
+done
+
+
+# Expected name of the container
+CONTAINER_NAME="cc-$ARCH-$OS-$RMW-$DISTRO"
+CONTAINER_TAG="$USER/$ARCH-$OS-$RMW-$DISTRO:latest"
+setup
+trap 'cleanup $CONTAINER_NAME $RESULT' EXIT
```

Comment:
line 76 and 77 should be inversed, as if the setup fails the cleanup may end up not being called.